### PR TITLE
Well known protos

### PR DESCRIPTION
**Pull request by antaljanosbenjamin**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add support for the [well-known Protobuf types](https://protobuf.dev/reference/protobuf/google.protobuf/) in the Protobuf format.

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 4e780fbfcf6cb3f62613e8975dfe468eeb9d90d2 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/56741/4e780fbfcf6cb3f62613e8975dfe468eeb9d90d2/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6864085305">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/56741/4e780fbfcf6cb3f62613e8975dfe468eeb9d90d2/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: contrib/google-protobuf-cmake/CMakeLists.txt [hunk 1]
```diff
@@ -317,9 +320,25 @@ endif ()
...
+  ${protobuf_source_dir}/src/google/protobuf/struct.proto
+  ${protobuf_source_dir}/src/google/protobuf/timestamp.proto
+  ${protobuf_source_dir}/src/google/protobuf/type.proto
+  ${protobuf_source_dir}/src/google/protobuf/wrappers.proto
+)
+
 add_library(_protobuf INTERFACE)
 target_link_libraries(_protobuf INTERFACE _libprotobuf)
 target_include_directories(_protobuf INTERFACE "${Protobuf_INCLUDE_DIR}")
+set_target_properties(_protobuf PROPERTIES INTERFACE_well_known_proto_files "${well_known_proto_files}")
```
Comment (Felixoid) [inline contrib/google-protobuf-cmake/CMakeLists.txt:341]: Maybe, `google_proto_files` reflects the content better?

**Review by Felixoid — APPROVED** [review]
```
> dpkg-deb -c clickhouse-common-static_23.11.1.1024_amd64.deb 
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/bin/
-rwxr-xr-x root/root 502549664 2023-11-15 15:12 ./usr/bin/clickhouse
-rwxr-xr-x root/root  20115202 2023-11-15 15:13 ./usr/bin/clickhouse-diagnostics
l--------- 0/0               0 2023-11-15 15:12 ./usr/bin/clickhouse-extract-from-config -> clickhouse
-rwxr-xr-x root/root 194059984 2023-11-15 15:00 ./usr/bin/clickhouse-library-bridge
-rwxr-xr-x root/root 194772296 2023-11-15 15:00 ./usr/bin/clickhouse-odbc-bridge
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/bash-completion/
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/bash-completion/completions/
-rw-r--r-- root/root       959 2023-11-15 14:41 ./usr/share/bash-completion/completions/clickhouse
-rw-r--r-- root/root       154 2023-11-15 14:41 ./usr/share/bash-completion/completions/clickhouse-benchmark
-rw-r--r-- root/root      5490 2023-11-15 14:41 ./usr/share/bash-completion/completions/clickhouse-bootstrap
-rw-r--r-- root/root       151 2023-11-15 14:41 ./usr/share/bash-completion/completions/clickhouse-client
-rw-r--r-- root/root       150 2023-11-15 14:41 ./usr/share/bash-completion/completions/clickhouse-local
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/clickhouse/
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/clickhouse/proto/
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/clickhouse/proto/google/
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/clickhouse/proto/google/protobuf/
-rw-r--r-- root/root      5909 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/any.proto
-rw-r--r-- root/root      7734 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/api.proto
-rw-r--r-- root/root     38497 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/descriptor.proto
-rw-r--r-- root/root      4895 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/duration.proto
-rw-r--r-- root/root      2363 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/empty.proto
-rw-r--r-- root/root      8185 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/field_mask.proto
-rw-r--r-- root/root      2341 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/source_context.proto
-rw-r--r-- root/root      3779 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/struct.proto
-rw-r--r-- root/root      6459 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/timestamp.proto
-rw-r--r-- root/root      6126 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/type.proto
-rw-r--r-- root/root      4042 2023-11-15 14:42 ./usr/share/clickhouse/proto/google/protobuf/wrappers.proto
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/doc/
drwxr-xr-x root/root         0 2023-11-15 15:14 ./usr/share/doc/clickhouse-common-static/
-rw-rw-r-- root/root       203 2023-11-15 14:41 ./usr/share/doc/clickhouse-common-static/AUTHORS
-rw-rw-r-- root/root    401959 2023-11-15 14:41 ./usr/share/doc/clickhouse-common-static/CHANGELOG.md
-rw-rw-r-- root/root     11389 2023-11-15 14:41 ./usr/share/doc/clickhouse-common-static/LICENSE
-rw-rw-r-- root/root      4446 2023-11-15 14:41 ./usr/share/doc/clickhouse-common-static/README.md
```

Looks good!

**Comment by antaljanosbenjamin** [comment]
Test failures:
- `Integration tests (asan) [4/4]` #55168
- `Stateless tests (release, analyzer)`
  - `01600_parts_states_metrics_long` #57307
  - `01238_http_memory_tracking` flaky #57316

**Review by Felixoid — APPROVED** [review]
Unfortunately, I don't get the improvements, so it's good from the packaging perspective.

For the code, I think it's the best to get another approval

**Review by antonio2368 — APPROVED** [review]

**Pull request merged**